### PR TITLE
LPS-70824 Control Panel elements are not responsive to role permission definition toggles

### PR DESCRIPTION
**Pull request by drewbrokke**
This is an update for [LPS-70824](https://issues.liferay.com/browse/LPS-70824).<br><br>@jorgeferrer, the last commit is one possible way of making sure that after the new permission is added, it won't be re-added on a re-deploy if a user removed the permission manually.  It seems to work well, but I'd like you to take a look just in case there is some possible issue with using PortletPreferences (for example, how likely is it that we would run into a ReadOnlyException while saving the portletPreferences?).  Thanks!<br><br>/cc @pei-jung

### CODE REVIEW

**Comment by drewbrokke** [comment]
ci:stop

**Comment by drewbrokke** [comment]
ci:retest

**Comment by JorgeFerrer** [comment]
Hey Drew,

This can work, although it's a new pattern and this is a pretty common need. I think we need to check if there is an existing pattern to do it. If not we should come up with one that is a bit more generic.

Could you bring it up to Bchan just in case he has any specific idea about it?

File: modules/apps/web-experience/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java [hunk 1]
```diff
@@ -308,17 +340,24 @@ protected void initPersonalControlPanelPortletPermission(
...
+
+			portletPreferences.setValue(
+				_MY_ACCOUNT_PERMISSIONS_INITIALIZED, StringPool.TRUE);
+
+			portletPreferences.store();
 		}
 
+		private static final String _MY_ACCOUNT_PERMISSIONS_INITIALIZED =
+			"myAccountPermissionsInitialized";
+
```
Comment (drewbrokke) [inline modules/apps/web-experience/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java]: @brianchandotcom Jorge asked me to check with you on this pattern for insuring that this code is only run once per portlet instance.  Do we have an existing pattern somewhere else to accomplish this?

Comment (drewbrokke) [inline modules/apps/web-experience/application-list/application-list-api/src/main/java/com/liferay/application/list/PanelAppRegistry.java]: @brianchandotcom here is the larger context: https://github.com/JorgeFerrer/liferay-portal/pull/812

**Pull request closed without merging**